### PR TITLE
Fix #8

### LINKED ISSUE #8: Humidifier entity will turn off by itself
image

It seems like to work properly but has a problem... When the sensor value reaches target humidity, it turns off the smart plug successfully, and it will turn off by itself in few seconds.

I know there's already a logic handles internal state change. But I still don't know why this won't work.
https://github.com/adrianByv/home-assistant-custom-components-switch-humidifier/blob/b4d7b5c7eda1b4aaf1773c8fb4b780f2807b55d8/custom_components/switch_humidifier/humidifier.py#L216-L226

### PR DESCRIPTION
**Pull request by auspic7**
This commit will add new condition checks if the power state of switch is changed.
fixes #8.

Please check and merge!

### CODE REVIEW

File: custom_components/switch_humidifier/humidifier.py [hunk 1]
```diff
@@ -205,7 +205,8 @@ def turn_off(self, **kwargs):
 
   def _state_changed(self, entity_id, old_state, new_state):
     """Called on sensor or switch state change"""
-    if not new_state is None and not new_state.state == 'unknown' and not new_state.state == 'unavailable':
+    if not new_state is None and not new_state.state == 'unknown' and not new_state.state == 'unavailable' and old_state.state != new_state.state:
+      # old switch power and new switch power must be different
```
Comment (ApoAdrian) [inline custom_components/switch_humidifier/humidifier.py]: This method is for any state change (humidity sensor or switch), not only switch changes.
Can you update the comment to something like "Check if they are different states"

File: custom_components/switch_humidifier/humidifier.py [hunk 2]
```diff
@@ -205,7 +205,8 @@ def turn_off(self, **kwargs):
 
   def _state_changed(self, entity_id, old_state, new_state):
     """Called on sensor or switch state change"""
-    if not new_state is None and not new_state.state == 'unknown' and not new_state.state == 'unavailable':
+    if not new_state is None and not new_state.state == 'unknown' and not new_state.state == 'unavailable' and old_state.state != new_state.state:
```
Comment (ApoAdrian) [inline custom_components/switch_humidifier/humidifier.py]: Can you use the same way of checking as in the rest, "not old_state.state == new_state.state" or replace the others "new_state.state != 'unavailable'".
Just  for unifying it.

**Review by ApoAdrian — COMMENTED** [review]
Thanks for finding and fixing the bug. Can you make this little changes? I'll merge it directly.

**Comment by auspic7** [comment]
Thank you for your review! I made a new commit based on your comment!

**Pull request merged**